### ISSUE TITLE
TimberPost picks up the wrong post ID in preview mode

### ISSUE THREAD

**Issue by bfred-it** [timber/timber#1318]
### Steps to reproduce behavior

1. Publish a post in the admin
2. `global $post->ID === 1` `new TimberPost()->ID === 1`
3. Make change in post
4. Click "Preview Changes" button
5. `global $post->ID === 1` `new TimberPost()->ID === 2` <-- mismatch

### Expected behavior

`new TimberPost()->ID` should be the same as `global $post; $post->ID`, which is the original post's ID

### Actual behavior

`new TimberPost()->ID` is the revision/preview ID, not the original post's ID.

This causes `new TimberPost()->thumbnail()` to return nothing if the thumbnail was changed.

This fixes it for me:

```php
global $post;
new TimberPost($post->ID);
```

### What version of WordPress, PHP and Timber are you using?
PHP 5.5.37
WordPress 4.7.1
Timber 1.2.1

### How did you install Timber? (for example, from GitHub, Composer/Packagist, WP.org?)

Composer

**Comment by daltonrooney** [comment]
I was using `$post = new TimberPost()` to get the post in my page.php template, which caused the wrong template to be returned for previews. I'm using something along the lines of 

`Timber::render( array( 'layouts/' . $post->post_name . '.twig', 'layouts/default.twig' ), $context );`

to return the template. The slug should have been 'home' but TimberPost was returning '1983-autosave-v1', so default.twig was returned instead of home.twig.

Quick and easy fix was pass the ID of the current page to TimberPost like

`$post = new TimberPost(get_the_ID());`

**Comment by mgussekloo** [comment]
Same problem here! Please advise.

**Comment by gchtr** [comment]
Is there a difference if you use

```php
$post = Timber::get_post();
```

instead of `$post = new TimberPost();`?

**Comment by sarukuku** [comment]
@bfred-it why was this issue closed?

**Comment by bfred-it** [comment]
Mistake 😅

**Comment by luigitec** [comment]
HI @bfred-it I think It's not an issue. 

The thing is that on the **preview mode**, it returns the post "revision" ID instead of the actual post ID, to retrieve such ID: you can do it requesting `post_parent`, that is `new TimberPost()->post_parent`.

**Comment by bfred-it** [comment]
No, it should be the post id like Wordpress naturally does. I shouldn’t get its parent.

**Comment by dreamon11** [comment]
Same problem here.

PHP 7.2.5
WordPress 4.9.6
Timber 1.7.1

Any ideas? :/

**Comment by dreamon11** [comment]
@bfred-it: do you have the problem too?

**Comment by dreamon11** [comment]
Same problem here.

PHP 7.2.5
WordPress 4.9.6
Timber 1.7.1

The fix from @bfred-it doesn't work anymore for me.

When I try, in single.php:

```
global $post;
var_dump($post->ID); // returns 29
$post = new Timber\Post($post->ID);
var_dump($post->ID); // return 12095
die;
```

Any ideas?

**Comment by dreamon11** [comment]
I managed to solve the problem:

- Downgrade from 1.7.1 to 1.7.0
- Replace `$post = new Timber\Post();` by `$post = Timber\Timber::query_post();`

It doesn't work with 1.7.1, with 1.7.0 it's OK.

I think it's because of changes in lib/Post.php.

**Comment by jarednova** [comment]
Closed by #1752 #1809
